submissions/BlueLamantine/oop/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/BlueLamantine/a-tiny-JS-world
+   Web app: https://bluelamantine.github.io//a-tiny-JS-world/
+   */
+  class General {
+   constructor({ name, gender, friends }) {
+     this.name = name;
+     this.gender = gender;
+     this.friends = Array.isArray(friends) ? friends : 'no friends';
+   }
+ }
+ 
+ class Human extends General {
+   constructor({ name, gender, friends, voise }) {
+     super({ name, gender, friends });
+     this.voise = voise;
+     this.species = 'human';
+     this.legs = 2;
+     this.hands = 2;
+   }
+ 
+   getGender() {
+     return {
+       male: ['He', 'his'],
+       female: ['She', 'her'],
+     };
+   }
+ 
+   getInfo() {
+     return (
+       `This ${this.species} is a ${this.gender}, ` +
+       `${this.getGender()[this.gender][1]} name is ${this.name}. ` +
+       `${this.getGender()[this.gender][0]} has ${this.legs} legs and ${
+         this.hands
+       } hands. ` +
+       `${this.getGender()[this.gender][0]} says \'${
+         this.voise
+       }\'. Friend with : ${this.friends}`
+     );
+   }
```

Comment:
A class should handle the properties it owns immediately. There must be a good reason to handle a class' property in any of its children.

Use method overloading and `super.methodName` calls where required.

Reply:
Ok